conda_build/skeletons/cpan.py

```diff
@@ -571,32 +620,74 @@ def dist_for_module(cpan_url, module, perl_version, config):
     distribution.
     '''
     # First check if its already a distribution
+    rel_dict = release_module_dict(cpan_url, module)
+    if rel_dict is not None:
+        distribution = module
+    else:
+        # Check if it's a module instead
+        mod_dict = core_module_dict(cpan_url, module)
+        distribution = mod_dict['distribution']
+
+    return distribution
+
+
+def release_module_dict(cpan_url, module):
     try:
-        rel_dict = get_cpan_api_url('{0}/release/{1}'.format(cpan_url, module), colons=False)
+        rel_dict = get_cpan_api_url(
+            '{0}/release/{1}'.format(cpan_url, module), colons=False)
     # If there was an error, module may actually be a module
     except RuntimeError:
         rel_dict = None
     except CondaHTTPError:
         rel_dict = None
-    else:
-        distribution = module
 
-    # Check if it's a module instead
-    if rel_dict is None:
-        try:
-            mod_dict = get_cpan_api_url('{0}/module/{1}'.format(cpan_url, module), colons=True)
+    return rel_dict
+
+
+def core_module_dict(cpan_url, module):
+    try:
+        mod_dict = get_cpan_api_url(
+            '{0}/module/{1}'.format(cpan_url, module), colons=True)
         # If there was an error, report it
-        except CondaHTTPError as exc:
-            core_version = core_module_version(module, perl_version, config=config)
-            if core_version is None:
-                sys.exit(('Error: Could not find module or distribution named'
-                          ' %s on MetaCPAN. Error was: %s') % (module, str(exc)))
-            else:
-                distribution = 'perl'
+    except CondaHTTPError:
+        sys.exit(('Error: Could not find module or distribution named'
+                  ' %s on MetaCPAN. Error was: %s') % (module))
+    else:
+        mod_dict = {}
+        mod_dict['distribution'] = 'perl'
+
+    return mod_dict
+
+
+@memoized
+def metacpan_api_is_core_version(cpan_url, module):
+
+    url = '{0}/release/{1}'.format(cpan_url, module)
+    url = url.replace("::", "-")
```

Comment:
`metacpan_api_is_core_version()` only works if the module->distribution mapping is a simple `s/::/-/g` but that doesn't hold true for all modules, namely `YAML::XS` which kept it's old distribution name of `YAML-LibYAML` (see https://fastapi.metacpan.org/v1/module/YAML::XS).

I'm going to open an issue and submit a PR for this.